src/net_processing.cpp

```diff
@@ -1406,6 +1409,62 @@ void PeerManagerImpl::FindNextBlocksToDownload(const Peer& peer, unsigned int co
     }
 }
 
+void PeerManagerImpl::TryDownloadingHistoricalBlocks(const Peer& peer, unsigned int count, std::vector<const CBlockIndex*>& vBlocks, const CBlockIndex *from_tip, const CBlockIndex* target_block)
+{
+    if (vBlocks.size() >= count) {
+        return;
+    }
+
+    vBlocks.reserve(count);
+    CNodeState *state = State(peer.m_id);
+    assert(state != nullptr);
+
+    if (state->pindexBestKnownBlock == nullptr || state->pindexBestKnownBlock->GetAncestor(target_block->nHeight) != target_block) {
+        // This peer is useless to us -- it has a chain that doesn't contain
+        // our assumeutxo target.
+        // Presumably this peer's chain has less work than our ActiveChain()'s tip, or else we
+        // will eventually crash when we try to reorg to it. Let other logic
+        // deal with whether we disconnect this peer.
+        return;
+    }
+
+    const CBlockIndex *pindexWalk = from_tip;
+
+    int nWindowEnd = pindexWalk->nHeight + BLOCK_DOWNLOAD_WINDOW;
+    int nMaxHeight = std::min<int>(target_block->nHeight, nWindowEnd + 1);
+
+    std::vector<const CBlockIndex*> vToFetch;
+
+    while (pindexWalk->nHeight < nMaxHeight) {
+        int nToFetch = std::min(nMaxHeight-pindexWalk->nHeight, std::max<int>(count-vBlocks.size(), 128));
```

Comment:
In commit "First draft of background sync logic" (6be9fd22ee2b9759cdd30a7bca4d07a9346eb57b)

This code is just copied and pasted from previous code which was added back in #4468, but `std::max` seems like a bug here, and I would expect it to be `std::min` if the point is to read up to 128 successors of pIndexWalk at a time.